app/src/main/kotlin/photos/network/home/photos/PhotosScreen.kt

```diff
@@ -138,8 +143,23 @@ fun PhotosContent(
     uiState: PhotosUiState,
     handleEvent: (event: PhotosEvent) -> Unit,
 ) {
-    // TODO: check permission first
-    handleEvent(PhotosEvent.StartLocalPhotoSyncEvent)
+    val lifecycleOwner = LocalLifecycleOwner.current
+    DisposableEffect(lifecycleOwner) {
+        val observer = LifecycleEventObserver { source, event ->
+            if (event == Lifecycle.Event.ON_PAUSE) {
+                // onPause
+                // TODO: stop observing media store
+            } else if (event == Lifecycle.Event.ON_RESUME) {
+                // onResume
+                handleEvent(PhotosEvent.StartLocalPhotoSyncEvent)
```

Comment:
what about the permision check you mentioned in that todo?

Reply:
This is already done here:
https://github.com/photos-network/android/blob/development/app/src/main/kotlin/photos/network/home/photos/PhotosScreen.kt#L70

Reply:
good boy 🤓